packages/remark-lint-no-undefined-references/index.js

```diff
@@ -8,9 +8,10 @@
  * The following options (default: `undefined`) are accepted:
  *
  * *   `Object` with the following fields:
- *     *   `allow` (`Array<string>`, default: `[]`)
- *         — text that you want to allowed between `[` and `]` even though it’s
- *         undefined
+ *     *   `allow` (`Array<string | RegExp | { source: string }>`,
+ *         default: `[]`)
+ *         — text or a regex that you want to be allowed between `[` and `]`
+ *         even though it’s undefined
```

Comment:
- Needs to explain the `source` object version
- Probably also needs to explain how regexes are handled (case-insensitive?)

Reply:
See db695ae40f39dd1014656cc9eb6792c473077683. WDYT?